components/tabs/tabs.js

```diff
@@ -393,11 +397,30 @@ class Tabs extends LocalizeCoreElement(ArrowKeysMixin(SkeletonMixin(LitElement))
 		return this.shadowRoot.querySelector('.d2l-tabs-container-list').getBoundingClientRect();
 	}
 
+	hideTab(tab) {
+		tab.setAttribute('data-state', 'removing');
+		return (Object.keys(this._tabIds).length > 1 && !reduceMotion) ? this._animateTabRemoval(tab) : Promise.resolve();
+	}
+
 	#checkTabPanelMatchRequested;
 	#panels;
 	#updateAriaControlsRequested;
 
-	_animateTabAddition(tabInfo) {
+	_animateTabAddition(tab) {
+		return new Promise((resolve) => {
+			const handleTransitionEnd = (e) => {
+				if (e.propertyName !== 'max-width') return;
+				if (tab) tab.removeEventListener('transitionend', handleTransitionEnd);
+				resolve();
+			};
+			if (tab) tab.addEventListener('transitionend', handleTransitionEnd);
+			tab.setAttribute('data-state', '');
```

Comment:
The code above checks if we have a `tab`, but then this line assumes it's defined. Should we check `if (!tab) return Promise.resolve();` at the beginning of this method?